_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-instagram.php

```diff
@@ -134,47 +138,11 @@ public function delete_instagram_access_token( $request ) {
 	/**
 	 * Get the Instagram Gallery.
 	 *
-	 * @param  Object $request - request passed from WP.
+	 * @param  WP_REST_Request $request The request.
 	 * @return mixed
 	 */
 	public function get_instagram_gallery( $request ) {
-		$site_id = $this->get_site_id();
-		if ( is_wp_error( $site_id ) ) {
-			return $site_id;
-		}
-
-		$path     = sprintf( '/sites/%d/instagram/%s?count=%d', $site_id, $request['access_token'], (int) $request['count'] );
-		$response = Client::wpcom_json_api_request_as_blog(
-			$path,
-			2,
-			array( 'headers' => array( 'content-type' => 'application/json' ) ),
-			null,
-			'wpcom'
-		);
-		if ( is_wp_error( $response ) ) {
-			return $response;
-		}
-		$body = json_decode( wp_remote_retrieve_body( $response ) );
-
-		return $body;
-	}
-
-	/**
-	 * Get the WPCOM or self-hosted site ID.
-	 *
-	 * @return mixed
-	 */
-	private function get_site_id() {
-		$is_wpcom = ( defined( 'IS_WPCOM' ) && IS_WPCOM );
-		$site_id  = $is_wpcom ? get_current_blog_id() : Jetpack_Options::get_option( 'id' );
-		if ( ! $site_id ) {
-			return new WP_Error(
-				'unavailable_site_id',
-				__( 'Sorry, something is wrong with your Jetpack connection.', 'jetpack' ),
-				403
-			);
-		}
-		return (int) $site_id;
+		return Jetpack_Instagram_Gallery_Helper::get_instagram_gallery( $request['access_token'], $request['count'] );
```

Comment:
Both params are marked as required in the route registration.
If you try calling the `/wpcom/v2/instagram/gallery` without params it returns 400 for `rest_missing_callback_param`.

I've also tried with empty params, but the route validation works well enough there as well.
- `?access_token&count`: error because `rest_invalid_param` (`count` is not numeric)
- `?access_token&count=1`: error because `rest_no_route`